### ISSUE TITLE
[Feature request] Remove zero size files with same name when moving

### ISSUE THREAD

**Issue by piranna** [mate-desktop/caja#1302]
When moving files from one folder to another (specially between devices, but not only), it shows a pop-up when both locations has folders or files with same names, asking if you want to replace or merge / don't move them, and allowing to do the same action for the next one, but for zero size files like old `.lock` files or `.media` files in Android folders, they could be safely (re)moved from the original place if they exist in the destination and are also zero size files because, well, there's no data at all :-) Same could be done for special files (pipes, character devices...), but I'm at this moment only interested on zero size files. This could allow to don't left in the original location a directories tree filled with zero size files.

#### MATE general version

1.20.4-2

#### Linux Distribution

Ubuntu Mate 19.04

**Comment by lukefromdc** [comment]
This can cause data loss in some cases where a zero lengh file (e.g a socket) needs to be retained, so this should not be done except manually. If anything at all is done with this feature request, it would need to be kept behind a popup dialog offering  "remove zero length files?"  or "copy zero length files"  with a default of "copy" and the dialog appearing only if a zero length file is detected. For a copy operation add "ignore zero length files". Note that you cannot actually "move" files between devices as that is essentially a renaming operation (same name, new path) and to "move" files to a new device actually means to copy to the new location, then delete from the old location.

I for one do not see this as a priority with a small team, lots of open PR's and lots of open bugs (this being a feature request with some complexity to avoid data loss).

**Comment by piranna** [comment]
> This can cause data loss in some cases where a zero lengh file (e.g a socket) needs to be retained, so this should not be done except manually.

I agree, that's why I say this can be done only for regular files, that's what I'm interested about. Obviously, "special files" like sockets needs a special consideration :-)


> If anything at all is done with this feature request, it would need to be kept behind a popup dialog offering "remove zero length files?" or "copy zero length files" with a default of "copy" and the dialog appearing only if a zero length file is detected. For a copy operation add "ignore zero length files".

That's *exactly* the behaviour I'm asking about :-)


> Note that you cannot actually "move" files between devices as that is essentially a renaming operation (same name, new path) and to "move" files to a new device actually means to copy to the new location, then delete from the old location.

Yes, I know that moving files are not actually moving, that are a rename in the same filesystem or a copy + delete in differente filesystems.


> I for one do not see this as a high priority with a small team, lots of open PR's and lots of open bugs (this being a feature request with some complexity to avoid data loss).

Would you accept a PR? Where should I start? I have faced this problem several times and would like to help to move it forward...

**Comment by lukefromdc** [comment]
I myself am not interested in this but cannot speak for the whole team. I would advise
waiting for feedback from others before expending time (we have a LOT of open PRs now),
unless of course you want to use it yourself in local builds anyway. In that case submit a PR
 when you have it working to your satisfaction and people can have a look at it. Again, I can't
make any promises for the others (let's see if anyyone else responds) and I myself am 
absoulely hammered for time myself right now.

**Comment by piranna** [comment]
That's fine, thank you for being so honest. I myself have a lot of other projects that can't attend due to lack of time :-) Ok, I'll wait to have any response, or see if I'm boring enought at work to start it :-P

**Comment by piranna** [comment]
As a better alternative, instead of just checking for empty files, do the checking and omit copying (only remove from source) when files are identical, where zero size files are just only a subset, just only that's the one easiest and faster to check :-) To check if we are overwritting a file with its same content (that's dumb) we can check first for their sizes, partial md5 hashes, and byte-to-byte comparation, the same way [fdupes](https://en.wikipedia.org/wiki/Fdupes) command works. The main problem I see here is when working with networked filesystems, that can be slow or have costs, but maybe a file size limit can be used the same way it's done with thumbnails...
